src/main/java/dev/joopie/jambot/util/SpotifyLinkParser.java

```diff
@@ -0,0 +1,26 @@
+package dev.joopie.jambot.util;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class SpotifyLinkParser {
+    public static String extractSpotifyId(String url) {
+        // Regex to extract the ID part
+        url = removeSiParameter(url);
+        String regex = "open\\.spotify\\.com/track/([a-zA-Z0-9]+)";
+        Pattern pattern = Pattern.compile(regex);
```

Comment:
Dit is een no-go. Compileer regex en zet het in een private static final. Regex compileren is duur.